### PR TITLE
add travis dockerhub config

### PR DESCRIPTION
**Pull request by fredsmith**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,53 +1,66 @@
...
+  - python: 3.8
+    env: TOXENV=django22
 install:
-  - docker exec -t discovery bash -c 'sed -i "s|http://archive|http://us.archive|g" /etc/apt/sources.list' # US mirrors for speed
-  - docker exec -t discovery bash -c 'apt update && apt install -y --no-install-recommends firefox gettext'
-  - docker exec -t discovery bash -c 'sed -i "s/course_discovery.settings.devstack/course_discovery.settings.test/" /edx/app/discovery/discovery_env'
-  - docker exec -e TOXENV=$TOXENV -t discovery bash -c 'source /edx/app/discovery/discovery_env && cd /edx/app/discovery/discovery/ && make requirements'
-
+- docker exec -t discovery bash -c 'sed -i "s|http://archive|http://us.archive|g"
+  /etc/apt/sources.list'
```
Comment (mikix) [inline .travis.yml:40]: You dropped the ` # US mirrors for speed` comment

File: .travis.yml [hunk 2]
```diff
@@ -1,53 +1,66 @@
...
-  - docker exec -t discovery bash -c 'source /edx/app/discovery/discovery_env && cd /edx/app/discovery/discovery/ && coverage xml'
-  - codecov
+- pip install -U codecov
+- docker exec -t discovery bash -c 'source /edx/app/discovery/discovery_env && cd
+  /edx/app/discovery/discovery/ && coverage xml'
+- codecov
+env:
+  global:
+  - DOCKER_USERNAME=edxbuilder
+  - secure: aOQRuNnH7pVTs/2upmUvymKk0F54khd5p5v0n8dngPhx6TV8Uv2HPCBBOl9SqYt/NQ/PRWu5EnscYx+tuVpi+ilRHdTZSlo0tFb6UATF9ZYODHsKlL+739PkmHorVSPaQ9C1SoPklrwqlHVRqy3Q6SYaChNAAR/hF8YuSKus8eQE0c7haCmwm/r4BsIOCdH5E6iuUw7lUWBjwOZu0Rv79LfmHy/KG1QjFPiY99BZm0BjzQ04jyA+uMqmtJB1xyO1kia5lp9OyPAeCtgZPtW8LwhsRRwISLHjEEMvawEGIZYS//VTqvfZ9juUBD8ihz8MiPzhpFdN6cdLfdLDldjzR0sQWcWJXNVZmsFd0SOD6iliQ5YMy8Xp1KCy5n4T+r7jufXRbvaq1R2WsiSo7n2ekt7+9BIH1mf3xDdNtpaYyUK5IEjNO1zX4fs473LenMwM74Vo9RyM5k6G0Jdo0g682Yxdy/1w3T12dXIhCv92s7so6iyNCm0S0XPxyQIfH9N65Zc9qGVqpPfQE6AfaAYbDzmC6LrfTJkoIwGRiwLBaMuzrRfoKbDcnjsrU3vXp3XSdYtMGSHO4GPwaGXs/SvHpXRxiLJzntUZWhbUdDSMPofRknC8vRwie7RsybiGwF+FaSK9pBKoON+qT8r6s6lWpvYT1W0WQcoE+kcngTMX70I=
```
Comment (mikix) [inline .travis.yml:63]: This is some encrypted value?

File: Makefile [hunk 3]
```diff
@@ -115,3 +115,19 @@ docs:
 
 check_keywords: ## Scan the Django models in all installed apps in this project for restricted field names
 	python manage.py check_reserved_keywords --override_file db_keyword_overrides.yml
+docker_build:
```
Comment (mikix) [inline Makefile:119]: Can you add a space between these make targets?

File: Makefile [hunk 4]
```diff
@@ -115,3 +115,19 @@ docs:
...
+	docker build . -f Dockerfile -t openedx/course-discovery
+	docker build . -f Dockerfile --target newrelic -t openedx/course-discovery:latest-newrelic
```
Comment (mikix) [inline Makefile]: I haven't messed with docker building much or targets. But it looks to me like our Dockerfile uncondtionally installs newrelic? What is the intent of `--target=newrelic` here?

Comment (fredsmith) [reply Makefile]: we build two docker tagged images, one with new relic, one without.    `--target newrelic` says to build all the way to the second entrypoint, which I just realized was misnamed.  Follow up fixes the names in the dockerfile

File: Makefile (hunk 3 above)
Comment (fredsmith) [reply Makefile:119]: all set

File: .travis.yml (hunk 2 above)
Comment (fredsmith) [reply .travis.yml:63]: labelled

File: .travis.yml (hunk 1 above)
Comment (fredsmith) [reply .travis.yml:40]: replaced

**Review by mikix — APPROVED** [review]
This is not related to your fix, but I just noticed the Dockerfile installs node 8, not node 12. We should fix that at some point.

**Pull request closed without merging**